src/main/java/com/projecty/projectyweb/message/MessageService.java

```diff
@@ -61,26 +61,22 @@ public void updateSeenDate(Message message) {
 
     public void sendMessage(
             String recipientUsername,
-            @ValidMessage Message message,
+            Message message,
             BindingResult bindingResult,
             List<MultipartFile> multipartFiles
     ) throws BindException {
-        Optional<User> recipient = userRepository.findByUsername(recipientUsername);
-        User sender = userService.getCurrentUser();
 
-        if (!recipient.isPresent()) {
-            ObjectError objectError = new ObjectError("recipient", messageSource.getMessage("message.recipient.invalid", null, LocaleContextHolder.getLocale()));
-            bindingResult.addError(objectError);
-        } else if (sender == recipient.get()) {
-            ObjectError objectError = new ObjectError("recipient", messageSource.getMessage("message.recipient.yourself", null, LocaleContextHolder.getLocale()));
-            bindingResult.addError(objectError);
-        }
+        messageValidator.validate(message, bindingResult);
+
         if (bindingResult.hasErrors()) {
             throw new BindException(bindingResult);
         }
 
+        User recipient = userRepository.findByUsername(recipientUsername).get();
```

Comment:
This code can probably cause NullPointerException. You should use method isPresent()

Reply:
I think it will not happen since I checked in `MessageValidator` already if recipient is available or not.